Mailer/app/api/v1/Handlers/Mailers/Traits/ParseLogFilterConditionsTrait.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+
+use Nette\Database\Table\Selection;
+
+trait ParseLogFilterConditionsTrait
+{
+    public function parseConditions(array $filter){
+
+        $conditions = [];
+
+        // DB column can be either key in case of array input or value in case of object input
+
+        foreach ($filter as $key => $value){
```

Comment:
* Please parse the `from`/`to` values as `DateTime` before using them as a value and change the time on the DateTime object

* Do not use `$key` directly as a column in the database, map it explicitly and throw an exception if something unexpected comes as a `$key`. If someone wants to use this trait, he/she should be capable of recognizing what "keys" are supported just by reading this trait. And if someone passes in `$filter` with invalid keys, he/she should know about it.

Reply:
This trait supports two formats, as is written in comment above. Therefore I will create an array of available columns and in if in_array condition will check whether the column is allowed.

Reply:
I understand that it supports two formats, but you still should be explicit about which `keys` are supported and which aren't. Right now, as a user of this trait, to use it I need to be aware of `mail_logs` DB schema. That's not right.

This is similar to what we've discussed earlier in this PR. User-input (even if it's coming from the validated API) shouldn't be able to get directly into the DB query. There needs to be a place (usually presenters/api handlers) which explicitly maps it and fails with understandable error if something is not right.

Reply:
Yes, I agree, I only wanted to clarify why I chose to use in_array check instead of map in the solution of this issue :).